services/slam/slam.go

```diff
@@ -110,6 +111,12 @@ func (svc *reconfigurableSlam) GetMap(ctx context.Context,
 	return svc.actual.GetMap(ctx, name, mimeType, cp, include, extra)
 }
 
+func (svc *reconfigurableSlam) GetInternalState(ctx context.Context, name string) ([]byte, error) {
+	svc.mu.RLock()
+	defer svc.mu.RUnlock()
+	return svc.actual.GetInternalState(ctx, name)
```

Comment:
This locking is used not only to prevent multiple reconfigures from interfering with one another but because if a reconfigure is necessary, it will wait until any ongoing function calls have been complete.

Therefore if say a `GetMap` call is on going and the config changes triggering a reconfiguration, that reconfiguration will not occur until the `GetMap` call is finished. Note: This may relate to the ticket regarding why reconfiguration is failing